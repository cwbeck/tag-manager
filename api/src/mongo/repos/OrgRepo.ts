import Repo from '../abstractions/Repo';
import { injectable } from 'inversify';
import Org from '../models/Org';
import { IndexSpecification } from 'mongodb';

@injectable()
export default class OrgRepo extends Repo<Org> {
    protected readonly auditEnabled = true;

    protected readonly indexes: IndexSpecification[] = [
        {
            background: false,
            key: {
                _name: 1,
            },
            unique: true,
        },
    ];
}
