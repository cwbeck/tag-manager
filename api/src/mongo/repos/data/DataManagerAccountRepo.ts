import Repo from '../../abstractions/Repo';
import { injectable } from 'inversify';
import DataManagerAccount from '../../models/data/DataManagerAccount';
import Org from '../../models/Org';
import userMessages from '../../../errors/UserMessages';
import GQLError from '../../../errors/GQLError';
import { AccountType } from '../../../enums/AccountType';

@injectable()
export default class DataManagerAccountRepo extends Repo<DataManagerAccount> {
    protected readonly auditEnabled = true;

    public async getFromOrg(org: Org): Promise<DataManagerAccount> {
        const account = await this.findOne({
            _org_id: org.id,
        });
        if (account === null) {
            throw new GQLError(userMessages.noDMAccount, true);
        }
        return account;
    }
    public async getSystemAccountFromOrg(org: Org): Promise<DataManagerAccount> {
        return this.findOneThrows(
            {
                _org_id: org.id,
                _account_type: AccountType.SYSTEM,
            },
            userMessages.accountFailed,
        );
    }
}
