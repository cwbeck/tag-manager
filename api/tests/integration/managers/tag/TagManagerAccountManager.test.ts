import Mock = jest.Mock;
import ManagerTestUtils from '../../../utils/ManagerTestUtils';
import TestUtils from '../../../utils/TestUtils';
import DataManagerAccountRepo from '../../../../src/mongo/repos/data/DataManagerAccountRepo';
import OrgRepo from '../../../../src/mongo/repos/OrgRepo';

describe('TagManagerAccountManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('fetches tag manager account', async () => {
        ManagerTestUtils.mockOrgUserWithViewAccess();

        const mockOrg = TestUtils.buildMockOrg();
        const mockOrgRepo = {} as OrgRepo;
        mockOrgRepo.findByIdThrows = jest.fn().mockResolvedValue(mockOrg);
        const mockSutModel = TestUtils.buildTagManagerAccount();
        const mockSutRepo = {} as DataManagerAccountRepo;
        mockSutRepo.findByIdThrows = jest.fn().mockResolvedValue(mockSutModel);

        (ManagerTestUtils.mockRepoFromModelFactory as Mock).mockImplementation((model: any) => {
            if (model.name === 'Org') {
                return mockOrgRepo;
            }
            return mockSutRepo;
        });

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                getTagManagerAccount(id: "5edf4c48e9615d9a1c66bc2a") {
                    id
                    org {
                        id
                        name
                    }
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
