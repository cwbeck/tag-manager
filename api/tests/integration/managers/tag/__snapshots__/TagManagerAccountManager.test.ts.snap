// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagManagerAccountManager fetches tag manager account 1`] = `
Object {
  "data": Object {
    "getTagManagerAccount": Object {
      "id": "5edf4c48e9615d9a1c66baee",
      "org": Object {
        "id": "5edf4c48e9615d9a1c66bbbb",
        "name": "testOrg",
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
